types/index.d.ts

```diff
@@ -1,4 +1,9 @@
-import Vue, { VNodeData, Component, ComponentOptions, FunctionalComponentOptions } from 'vue'
+import Vue, { VNodeData, ComponentOptions, FunctionalComponentOptions } from 'vue'
+
+export type Component =
```

Comment:
If `Component` type from vue package fails, it probably means we need to fix it in the main repo. Would you like to point out the failing usage?

Reply:
Indeed. I found the `Component` type already has the base `Vue` type for `this` type of `Data` and `Method` in default. That fails type checking if `ComponentOptions` has different type: https://github.com/vuejs/vue-test-utils/blob/dev/types/test/resources.ts#L9

Maybe we should change the default type parameters of `Component` (and `AsyncComponent`) to `{}`?

Reply:
Shall we merge this now and update the Component type here later when it's fixed in the main repo?

Reply:
Let's merge it now. We can change it later after new Vue is released.

Reply:
@ktsn The problem is that `Component` isn't correctly setting up type bound. 

`ComponentOption<V>` is contravariant, that is, `ComponentOption<Normal>` isn't a sub type of `ComponentOption<Vue>`, but of `ComponentOption<never>`.

`beforeCreate` in componentOption would also break assignment here if contravariant method was checked. But currently it is not https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#note

Reply:
@HerringtonDarkholme Thanks for clarify the problem! I didn't realize we can use `never` for the default generic type.

I'll change the `ComponentOptions`'s 1st type parameter to `never` for now before this PR is merged.